src/MsgPack/Serialization/SerializationTarget.cs

```diff
@@ -308,37 +307,59 @@ private static void VerifyKeyUniqueness( IList<SerializingMember> result )
 				}
 				catch ( ArgumentException )
 				{
-					List<MemberInfo> list;
+					List<SerializingMember> list;
 					if ( duplicated.TryGetValue( member.Contract.Name, out list ) )
 					{
-						list.Add( member.Member );
+						list.Add( member );
 					}
 					else
 					{
-						duplicated.Add( member.Contract.Name, new List<MemberInfo> { existents[ member.Contract.Name ].Member, member.Member } );
+						duplicated.Add( member.Contract.Name, new List<SerializingMember> { existents[ member.Contract.Name ], member } );
 					}
 				}
 			}
 
 			if ( duplicated.Count > 0 )
 			{
-				throw new InvalidOperationException(
-					String.Format(
-						CultureInfo.CurrentCulture,
-						"Some member keys specified with custom attributes are duplicated. Details: {{{0}}}",
-						String.Join(
-							",",
-							duplicated.Select(
-								kv => String.Format(
-									CultureInfo.CurrentCulture,
-									"\"{0}\":[{1}]",
-									kv.Key,
-									String.Join( ",", kv.Value.Select( m => String.Format( "{0}.{1}({2})", m.DeclaringType, m.Name, ( m is FieldInfo ) ? "Field" : "Property" ) ).ToArray() )
-								)
-							).ToArray()
-						)
-					)
-				);
+                foreach (KeyValuePair<string, List<SerializingMember>> kvp in duplicated)
+                {
+                    foreach (var memberInfo in kvp.Value)
+                    {
+                        if (memberInfo.Member.DeclaringType != typeof (object))
+                        {
+                            foreach (var derived in kvp.Value)
+                            {
+                                if (memberInfo.Member == derived.Member)
+                                {
+                                    continue;
+                                }
+
+                                if (memberInfo.Member.DeclaringType.IsSubclassOf(derived.Member.DeclaringType))
+                                {
+                                    result.Remove(memberInfo);
+                                }
+                            }
+                        }
+                    }
+                }
+
+//				throw new InvalidOperationException(
```

Comment:
i want to keep their original code so if we ever pull changes from upstream master we don't undo my change.